### PR TITLE
Bmesh mesh data

### PR DESCRIPTION
**Pull request by og76**
just like object mesh data

vert vgroups returns [] as vgroups are in object

to have vgroups "carried by bmesh" we can add a vert custom layer in bmesh from object

### CODE REVIEW

File: data_structures/mesh.py [hunk 1]
```diff
@@ -80,6 +87,16 @@ class Polygon:
     __slots__ = ("vertexLocations", "normal", "center", "area", "materialIndex")
 
     @staticmethod
+    def fromBMeshFace(bmeshFace, centerWeighted = False):
```
Comment (JacquesLucke) [inline data_structures/mesh.py]: why is there a centered weight parameter?

File: nodes/mesh/bmesh_mesh_data.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        self.newOutput("Vertex List", "Vertices", "vertices")
+        self.newOutput("Polygon List", "Polygons", "polygons")
+
+    def getExecutionCode(self):
+        isLinked = self.getLinkedOutputsDict()
+        if not any(isLinked.values()): return
+
+#        yield "meshName = ''"
+#        yield "if getattr(object, 'type', '') == 'MESH':"
+#        yield "    mesh = self.getMesh(object, useModifiers, scene)"
```
Comment (JacquesLucke) [inline nodes/mesh/bmesh_mesh_data.py]: you should remove code that is not used.

File: data_structures/mesh.py (hunk 1 above)
Comment (og76) [inline data_structures/mesh.py]: ah, I had another variant cause you can calc center as median or median weighted with an advanced option

I'll test that again to make sure we need it

File: nodes/mesh/bmesh_mesh_data.py (hunk 2 above)
Comment (og76) [inline nodes/mesh/bmesh_mesh_data.py]: sorry, I'll del all commented

**Comment by og76** [comment]
may remove the weigthed center variant if affects performance in any way
not sure how useful it really is, and the blender / mesh poly center is the other

**Pull request merged**